syntax = "proto3";

package TCProto;

import "map_renderer.proto";
import "transport_router.proto";
message Coordinates
{
    double lat = 1;
    double lng = 2;
}

message Distance
{
    bytes  firstStop = 1;
    bytes  secondStop = 2;
    uint32 distance = 3;
}

message Stop
{
    Coordinates coords = 1;
    bytes  name = 2;
    repeated bytes busRoutes = 3;
    uint32 id = 4;
}

message Bus
{
    bytes name = 1;
    repeated bytes stopBus = 2;
    uint32 stops_on_route = 3;
    uint32 unique_stops = 4;
    uint32 sum_fact_dist = 5;
    double curvature = 6;
    bool is_roundtrip = 7;
}

message TransportCatalogue
{
    repeated Stop stops = 1;
    repeated Distance distanceBetweenStops = 2;
    repeated Bus buses = 3;
}

message TransportData
{
    TransportCatalogue transportCatalogue = 1;
    rend.RenderSettings renderSettings= 2; 
    pr_route.RouterSettings routerSettings = 3;
}
